warehouse/internal/repo/schema_test.go

```diff
@@ -142,5 +142,50 @@ func TestWHSchemasRepo(t *testing.T) {
 		t.Log("empty")
 		_, err = r.GetTablesForConnection(ctx, []warehouseutils.SourceIDDestinationID{})
 		require.EqualError(t, err, errors.New("no source id and destination id pairs provided").Error())
+
+		t.Log("multiple")
+		latestNamespace := "latest_namespace"
+		sourceID2 := "source_id_2"
+		destinationID2 := "destination_id_2"
+		connection2 := warehouseutils.SourceIDDestinationID{SourceID: sourceID2, DestinationID: destinationID2}
+		schemaLatest := model.WHSchema{
+			UploadID:        2,
+			SourceID:        sourceID,
+			Namespace:       latestNamespace,
+			DestinationID:   destinationID,
+			DestinationType: destinationType,
+			Schema:          schemaModel,
+			CreatedAt:       now,
+			UpdatedAt:       now,
+		}
+		schema2 := model.WHSchema{
+			UploadID:        3,
+			SourceID:        sourceID2,
+			Namespace:       namespace,
+			DestinationID:   destinationID2,
+			DestinationType: destinationType,
+			Schema:          schemaModel,
+			CreatedAt:       now,
+			UpdatedAt:       now,
+		}
+		_, err = r.Insert(ctx, &schemaLatest)
+		require.NoError(t, err)
+		_, err = r.Insert(ctx, &schema2)
+		require.NoError(t, err)
+		expectedTableNames, err = r.GetTablesForConnection(ctx, []warehouseutils.SourceIDDestinationID{connection, connection2})
+		require.NoError(t, err)
+		require.Equal(t, len(expectedTableNames), 2)
+		require.Contains(t, expectedTableNames, warehouseutils.FetchTableInfo{
+			SourceID:      sourceID,
+			DestinationID: destinationID,
+			Namespace:     latestNamespace,
+			Tables:        []string{"table_name_1", "table_name_2"},
+		})
+		require.Contains(t, expectedTableNames, warehouseutils.FetchTableInfo{
+			SourceID:      sourceID2,
+			DestinationID: destinationID2,
+			Namespace:     namespace,
+			Tables:        []string{"table_name_1", "table_name_2"},
+		})
```

Comment:
You don't really need the `connection2` and `schema2`, right? I find it confusing since the assertion really is about something else here.

I think it would make more sense to check the whole slice (with `ElementsMatch` instead of `Contains`) once before insertion and then again after we insert the latest schema to make sure nothing changes because we already have an entry for that sourceID/destID pair.

What do you think? Something like this:

```suggestion
		t.Log("multiple")
		latestNamespace := "latest_namespace"
		expectedTableNames, err = r.GetTablesForConnection(ctx, []warehouseutils.SourceIDDestinationID{connection})
		require.NoError(t, err)
		require.ElementsMatch(t, expectedTableNames, []warehouseutils.FetchTableInfo{{
			SourceID:      sourceID,
			DestinationID: destinationID,
			Namespace:     latestNamespace,
			Tables:        []string{"table_name_1", "table_name_2"},
		}})
		schemaLatest := model.WHSchema{
			UploadID:        2,
			SourceID:        sourceID,
			Namespace:       latestNamespace,
			DestinationID:   destinationID,
			DestinationType: destinationType,
			Schema:          schemaModel,
			CreatedAt:       now,
			UpdatedAt:       now,
		}
		_, err = r.Insert(ctx, &schemaLatest)
		require.NoError(t, err)
		expectedTableNames, err = r.GetTablesForConnection(ctx, []warehouseutils.SourceIDDestinationID{connection})
		require.NoError(t, err)
		require.ElementsMatch(t, expectedTableNames, []warehouseutils.FetchTableInfo{{
			SourceID:      sourceID,
			DestinationID: destinationID,
			Namespace:     latestNamespace,
			Tables:        []string{"table_name_1", "table_name_2"},
		}}, "the new schema should not have changed the returned tables since we already had a schema for the same source id and destination id pair")
```

Reply:
@fracasula  thank you for reviewing. Yeah, I agree it makes the test more complicated. I was trying to test the output of many connections (source id - destination id pairs). 
I have updated the test only to test one connection. I don't think we need the test before inserting, because we do the same test before.